eskip/eskip.go

```diff
@@ -185,6 +179,85 @@ type RouteInfo struct {
 	ParseError error
 }
 
+// HasPredicate checks if the route has predicate with the given name.
+func (r *Route) HasPredicate(name string) bool {
+	for _, p := range r.Predicates {
+		if p.Name == name {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetPredicates returns all predicates that match the given name.
+func (r *Route) GetPredicates(name string) []*Predicate {
```

Comment:
similarly to HasPredicate, this function could be more generally useful and would better reflect what it does, if it worked on any slice of predicates, and not only on those that are on the same route. E.g:

```
func PredicatesByName(p []*Predicate, name string) []*Predicate {}
```